File: python/0038_CountAndSay.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        res='1'
        for i in range(n-1):
            tmp=''
            j=0
            while j<len(res):
                count=1
                while j+1<len(res) and res[j]==res[j+1]:
                    count+=1
                    j+=1
                tmp+=str(count)+res[j]
                j+=1
            res=tmp
        return res

File: cpp/0038_CountAndSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==1)
            return "1";
        string pre = countAndSay(n - 1);
        string res="";
        for(int i = 0; i < pre.size(); i++)
        {
            int count = 1;
            while(i+1 < pre.size() && pre[i] == pre[i+1]){
                i++;
                count++;
            }
            res+=(to_string(count)+pre[i]);
           
        }
        return res;

    }
    
};
